lib/event_sourcery/aggregate_root.rb

```diff
@@ -124,8 +124,12 @@ def load_history(events)
 
     def apply_event(event_class, options = {})
       event = event_class.new(**options.merge(aggregate_id: id))
-      mutate_state_from(event)
-      @changes << event
+      if event.valid?
```

Comment:
It won't work easily on `Event.new`, as events may be instantiated then modified before being stored. I'd like to discuss this PR and https://github.com/envato/event_sourcery/issues/139 at the next ES brown bag session. There's lots to think about here.